subsys/net/lib/coap/coap.c

```diff
@@ -972,6 +972,18 @@ int coap_block_transfer_init(struct coap_block_context *ctx,
 #define SET_MORE(v, m) ((v) |= (m) ? 0x08 : 0x00)
 #define SET_NUM(v, n) ((v) |= ((n) << 4))
 
+int coap_append_block_option(struct coap_packet *cpkt, struct coap_block_context *ctx)
+{
+	/* TODO this can be improved as the `is_request` check
+	 * is also performed in the called functions.
+	 */
+	if (is_request(cpkt)) {
```

Comment:
I don't actually think this is correct - both Block1 and Block2 options can be either part of request or response. The difference on the use of those options is whether you request data (GET) or proactively send (PUT/POST).

Reply:
Well, the function name doesn't suggest that. IMO if we provide such a generic function, it should cover all cases.

Reply:
I would prefer to name the function properly (i.e. `coap_append_descriptive_block_option`) instead of implementing a generic function that we don't need yet (YAGNI).